depthai_ros_driver/src/conversion.cpp

```diff
@@ -0,0 +1,107 @@
+#include "depthai_ros_driver/conversion.hpp"
+
+namespace rr {
+cv::Mat chw2hwc(const cv::Mat& mat) {
+    cv::Mat new_mat = cv::Mat(mat.size(), mat.type());
+    const auto WH = mat.cols * mat.rows;
+
+    const uchar* src = mat.data;
+    uchar* dst = new_mat.data;
+    for (int i = 0; i < WH; ++i) {
+        dst[3 * i + 0] = src[0 * WH + i];
+        dst[3 * i + 1] = src[1 * WH + i];
+        dst[3 * i + 2] = src[2 * WH + i];
+    }
+    return new_mat;
+}
+
+// prepare the cv size and cv type given dai::RawImgFrame::Type
+cv::Mat prepare_cv_mat_from_dai_type(dai::RawImgFrame::Type dai_type, int rows, int cols, int size) {
+    cv::Size cv_size = {0, 0};
+    int cv_type = 0;
+
+    switch (dai_type) {
+        case dai::RawImgFrame::Type::RGB888i:
+        case dai::RawImgFrame::Type::BGR888i:
+        case dai::RawImgFrame::Type::BGR888p:
+        case dai::RawImgFrame::Type::RGB888p:
+            cv_size = cv::Size(cols, rows);
+            cv_type = CV_8UC3;
+            break;
+
+        case dai::RawImgFrame::Type::YUV420p:
+        case dai::RawImgFrame::Type::NV12:
+        case dai::RawImgFrame::Type::NV21:
+            cv_size = cv::Size(cols, rows * 3 / 2);
+            cv_type = CV_8UC1;
+            break;
+
+        case dai::RawImgFrame::Type::RAW8:
+        case dai::RawImgFrame::Type::GRAY8:
+            cv_size = cv::Size(cols, rows);
+            cv_type = CV_8UC1;
+            break;
+
+        case dai::RawImgFrame::Type::GRAYF16:
+            cv_size = cv::Size(cols, rows);
+            cv_type = CV_16FC1;
```

Comment:
Woops, I cannot compile this code because opencv3 doesn't have `CV_16FC1`. So, this driver needs opencv greater than Opencv4.

Reply:
OpenCV4 and PCL 1.12 are recommended because of newer features :P

Reply:
Yup. I will update opencv of my environment. this is just an info.